### PR TITLE
Add featured products section

### LINKED ISSUE #1409: Add featured product list section

### PR DESCRIPTION
**Pull request by dhmacs**
closes #1409 

First of all, apologies for the big PR! 😬 To anyone who'll review this I'll owe you a pizza!
The bulk of the changes comes from ramifications of refactoring common concepts used by the product previews cards (e.g. money, price tiers, etc).

---

❗❗❗After this PR is merged Strapi should be redeployed in production ❗❗❗

cc @sterlinghirsh @danielbeardsley @masonmcelvain 

---

## Changelog

- Added a featured products section to the page template
- Polished `ProductGrid` and `ProductGridItem` which will be shared pretty much everywhere there's a grid of products (e.g. product lists, featured products section, etc). This refactor will happen in a follow up PR, for now the new `ProductGrid` and `ProductGridItem` are being used only in the featured products section
- Adding more models to centralize the logic of common entities (e.g. `Money`)

## QA

1. Open [store home page preview](https://react-commerce-git-add-featured-products-section-ifixit.vercel.app/store)
2. Verify that the featured products section is implemented according to [mockup](https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit---Working-files?node-id=10434%3A211428&t=1Q1yOcSyhFvpnYZq-4)

    Screenshot 2023-03-02 at 16 39 57

3. Verify that the section is responsive

> 👁️ We're just using dummy data and it can be edited from Strapi

### CODE REVIEW

**Comment by k0rvusk0r4x** [comment]
For the most part, this looks good.

When we have a product with a name that is one line long next to one that is two lines long, it looks a little weird. I would expect the stars to be lined up, and because of that, the space between the stars and price to be the same. I'm not sure if you can actually change that though. 
image

The mockup is five product tiles wide while this branch goes up to six product tiles wide.

dev_block 😿 

Cross browser and device tested

**Comment by dhmacs** [comment]
> When we have a product with a name that is one line long next to one that is two lines long, it looks a little weird. I would expect the stars to be lined up, and because of that, the space between the stars and price to be the same. I'm not sure if you can actually change that though. image

This is the behaviour we have in production. I'd say that if we want to discuss a different implementation, let's do it in a separate issue so we can move on with this PR

> The mockup is five product tiles wide while this branch goes up to six product tiles wide.

This is expected and informed by this [comment](https://github.com/iFixit/react-commerce/issues/1206#issuecomment-1425855195). I paired with @mmarcon91 to validate the responsive design, so the way it's behaving is expected, unless you notice some weird behaviour that we didn't spot.

un_dev_block 🟢

**Comment by k0rvusk0r4x** [comment]
This still working well and responding as it should.

QA 😺

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Great job! The parsing refactors looked good, I didn't notice any discrepancies from the previous implementations. My one comment does not need to be addressed by this PR

**Comment by masonmcelvain** [comment]
Redeployed strapi ✔️

File: frontend/components/sections/FeaturedProductsSection.tsx [hunk 1]
```diff
@@ -9,17 +9,25 @@ import { SectionHeading } from './SectionHeading';
 export interface FeaturedProductsSectionProps {
    title?: string | null;
    description?: string | null;
+   background?: BackgroundColor | null;
    products: ProductPreview[];
 }
 
+type BackgroundColor = 'transparent' | 'white';
```
[the remark this answers is not in the document]
Comment (dhmacs) [reply frontend/components/sections/FeaturedProductsSection.tsx:16]: Yes, great suggestion, I had missed this one 👍 I'll add it to the other PRs

**Pull request merged**